### ISSUE TITLE
DDP GPU0 allocate too many memory

### ISSUE THREAD

**Issue by yijingshihenxiule** [p0p4k/vits2_pytorch#36]
Hello, Thank you for your awesome work.  When I train with two gpus in recent two days, I find too many process in GPU0?
How can I deal with it?
![image](https://github.com/p0p4k/vits2_pytorch/assets/68212954/37015168-c483-495f-b3c1-9fc78f11aefe).

**Comment by p0p4k** [comment]
I am not sure why this is the case. If many people face this issue, I can try to look into it. Try to see what happens when using just 1 gpu (`CUDA_VISIBLE_DEVICES=1 train.py ...`)

**Comment by yijingshihenxiule** [comment]
Thank you for your answer. 
Maybe I find thereson. 
In https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#save-and-load-checkpoints, it says

> If map_location is missing, torch.load will first load the module to CPU and then copy each parameter to where it was saved, which would result in all processes on the same machine using the same set of devices.

But I tried to load checkpoint in cpu like vits1 and load checkpoint in GPU, both did not work for me.
Need Help!

**Comment by yijingshihenxiule** [comment]
> Thank you for your answer. Maybe I find thereson. In https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#save-and-load-checkpoints, it says
> 
> > If map_location is missing, torch.load will first load the module to CPU and then copy each parameter to where it was saved, which would result in all processes on the same machine using the same set of devices.
> 
> But I tried to load checkpoint in cpu like vits1 and load checkpoint in GPU, both did not work for me. Need Help!

Maybe the case has nothing to do with checkpoint load.
I am confused about it.

**Comment by p0p4k** [comment]
I'll keep this issue open. We can fix it later if it is causing any performance issues.

**Comment by p0p4k** [comment]
It is just a typo. The mask is being used in the transformers either way.

**Comment by yijingshihenxiule** [comment]
It was just my fault. I used wrong monotonic. Now it fixed. You can close this issue.
